Server/src/main/java/com/assignment/concurrent/domain/PointPairingTask.java

```diff
@@ -0,0 +1,77 @@
+package com.assignment.concurrent.domain;
+
+import java.sql.Time;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Random;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+import com.assignment.concurrent.service.PointsService;
+
+public class PointPairingTask implements Callable<Edge> {
+
+    private static int totalEdgeFormed;
+    private static long startTime;
+    private boolean threadFail;
+    private int failAttempts;
+    private int numberOfPoints;
+    private Point[] points;
+    private List<Edge> edgeArrayList;
+
+    public PointPairingTask(Point[] arr){
+        this.points = arr;
+        this.edgeArrayList = new ArrayList<Edge>();
+    }
+
+    @Override
+    public Edge call() throws Exception {
+        long start = 0;
+        long now = 0;
+        long target = 2;
+
+        while( (now-start) < target ){
+            edgeArrayList.add(formEdges());
+        }
+        
+        return null;
+    }
+
+    public int getEdgesFormed(){
+        return edgeArrayList.size();
+    }
+
+    public int getFailAttempts(){
+        return failAttempts;
+    }
+    
+    private Edge formEdges() throws InterruptedException {
+        Random r = new Random();
+        Edge edgeFormed = null;
+        int p1 = r.nextInt(points.length);
+        int p2 = r.nextInt(points.length);
+
+        // Try to acquire 2 points
+        while(points[p1].isLocked() || points[p2].isLocked() || (!points[p1].isHasEdge() && !points[p2].isHasEdge())) {
```

Comment:
`(!points[p1].isHasEdge() && !points[p2].isHasEdge())`
| p1 | p2 | (!p1 && !p2) |
|---|---|---|
| 0 | 0 | 1 | 
| 0 | 1 | 0 | 
| 1 | 0 | 0 | 
| 1 | 1 | 0 | 

@LUXIANZE , since both points must **NOT** have edge then only allow to enter the `while` loop.
what if one of the point has Edge? how can we `failAttempts++;`

Reply:
My bad, I should catch that somewhere else